docs/howto/how-to-make-tensorflow-plugin-with-ui/tensorflow-plugin-with-ui.md

```diff
@@ -0,0 +1,1488 @@
+
+```python
+    def on_record_batch(self, batch: "pa.Table", anchor: "Anchor") -> None:
+        """
+        Process the passed record batch.
+
+        The method that gets called whenever the plugin receives a record batch on an input.
+
+        This method IS NOT called during update-only mode.
+
+        Parameters
+        ----------
+        batch
+            A pyarrow Table containing the received batch.
+        anchor
+            A namedtuple('Anchor', ['name', 'connection']) containing input connection identifiers.
+        """
+        if self.MODE == "PREDICT":
+            model = tf.keras.saving.load_model(f'{self.provider.tool_config["modelConfig"]["modelName"]}-exported', custom_objects={'custom_standardization': custom_standardization})
+            self.info("Loaded model, predicting...")
+            
+            try:
+                results = model.predict(batch['Beep'].to_pylist())
+            except Exception as e:
+                self.provider.io.error(f"ERR during predict")
+                raise e
+                
+            df = pd.DataFrame({"Results": [float(n[0]) for n in results]})
+            batch_to_send = pa.RecordBatch.from_pandas(df=df, preserve_index=False)
+            self.provider.write_to_anchor("Output", batch_to_send)
+            self.info("TextClassifier tool done.")
+```
+Here, we load our model with `keras. saving.load_model(...)`. Note that we append `-exported' to differentiate our training and "production" models. 
+As you develop more serious tools, you can create custom methods to deploy and store model deployment and storage. Here, we note the ability to do so!
+
+Then, we call `model.predict(...)` on our loaded model, take the results, and write them to our output anchor for use in the workflow.
+
+Now, this is all of our backend code! Since our (soon-to-exist) tests are passing, we'll move on to our front end and enable these mocked values to be dynamically set by the User!
+
+
+## Writing the Frontend: UI SDK
+
+_Please Note: if you are unfamiliar with React and the UI SDK, we recommend the (tbd link)[Getting Started with UI SDK and PythonSDK] guide first or as a supplementary, as you may need to reference the above guide for detailed steps._
+
+As some quick prep work, let's make sure you set up your front end. As of (E-Note: version num of "no UI" option?)], if you did not initialize a UI as described in the guide linked above, you might not have one by default. No problem! Run:
+`ayx_plugin_cli generate-ui` (E-Note: Correct command?)
+Then you should then have your tool's UI files under `workspace-dir/ui/ToolName`. 
+Next, we want to start up the UI SDK's [dev-harness](https://alteryx.github.io/alteryx-ui/).
+You should be able to run `npm install` and then `npm start` as expected in a React/node app.
+If you hit an issue, we recommend referring to the recommended supplementary or other troubleshooting tips in detail on the `alteryx-ui` documentation.
```

Comment:
TW review:

As some quick prep work, set up your front end. As of (E-Note: version num of "no UI" option?)], if you did not initialize a UI as described in the guide linked above, you might not have one by default. No problem!

Run `ayx_plugin_cli generate-ui` (E-Note: Correct command?).

You should then have your tool's UI files under `workspace-dir/ui/ToolName`.  Next, we start up the UI SDK's [dev-harness](https://alteryx.github.io/alteryx-ui/). You should be able to run `npm install` and then `npm start` as expected in a React/node app. If you encounter an issue, we recommend that you refer to the recommended supplementary guide or other troubleshooting tips in detail on the `alteryx-ui` documentation.

Reply:
Resolution note: UI companion piece updated with `generate-ui` info